extension/lib/shopgate/CartExtensionPipeline.js

```diff
@@ -133,44 +171,22 @@ class ShopgateCartExtensionPipeline {
    */
   async updateProducts (cartItems) {
     const cartId = await this.getCartId()
-
     let updateSuccess = true
-    try {
-      await this._bigCommerceCartRepository.updateItems(
-        cartItems.map((item) => {
-          return BigCommerceCartRepository.createLineItemUpdate(item.cartItemId, item.quantity)
-        }),
-        async (failureEvent) => {
-          this._context.log.error(decorateDebug({
-            reason: failureEvent.reason,
-            cartItemId: failureEvent.item.itemId,
-            quantity: failureEvent.item.quantity
-          }), 'Failed updating product')
 
-          await this._messagesRepository.push(cartId, failureEvent.reason, failureEvent.item.itemId)
+    await this.updateItems(cartItems.map((item) => {
+      return BigCommerceCartRepository.createLineItemUpdate(item.cartItemId, item.quantity)
+    }),
+    async (failureEvent) => {
+      this._context.log.error(decorateDebug({
+        reason: failureEvent.reason,
+        cartItemId: failureEvent.item.itemId,
+        quantity: failureEvent.item.quantity
+      }), 'Failed updating product')
 
-          updateSuccess = false
-        }
-      )
-    } catch (err) {
-      if (err.code !== 422) {
-        await this._messagesRepository.push(cartId, 'We were unable to update the cart. Please try again later.')
-        throw err
-      }
+      await this._messagesRepository.push(cartId, failureEvent.reason, failureEvent.item.itemId)
 
-      // in bigc cart api details of the error are hidden inside of error message
-      const errorMessageMatch = err.message.match(/({.+})/)
-      let errorMessage = 'Items in your cart couldn\'t be updated. Please try again later.'
-      if (errorMessageMatch) {
-        try {
-          errorMessage = JSON.parse(errorMessageMatch[1]).title
-        } catch (err) {
-          this._context.log.error(decorateError(err), 'Unable to process the error from BigC api')
-        }
-      }
-
-      await this._messagesRepository.push(cartId, errorMessage)
-    }
+      updateSuccess = false
+    }, false)
```

Comment:
ugh, this may need discussion/peer review stuff. Why? Because of the logic you are changing, I thought we wanted to throw the ECART on every item update/add error. Isn't that per new pipeline specs?

Reply:
No, i only throw an error in case if we are in the product view. The reason for that is, that an throw error open an popup with the error message. We dont need that in the cart view, because the error message is shown on the top of the page in a message area. If i also throw there he error, we have both - Popup and in the message box.